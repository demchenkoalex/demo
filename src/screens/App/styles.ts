import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    flex: 1
  },
  text: {
    fontWeight: 'bold',
    paddingVertical: 16,
    textAlign: 'center'
  }
})
