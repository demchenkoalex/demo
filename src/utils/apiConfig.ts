import axios from 'axios'

const api = axios.create({
  baseURL: 'https://reqres.in/api',
  timeout: 10000
})

export default api
