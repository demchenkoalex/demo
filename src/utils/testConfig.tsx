import React, { ComponentType } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'

export const setupJestShallow = ({ Component, props }: { Component: ComponentType<any>, props: any }) => {
  const shallowRenderer = createRenderer()

  shallowRenderer.render(<Component {...props} />)

  const wrapper = shallowRenderer.getRenderOutput()
  const instance = shallowRenderer.getMountedInstance() as any

  return {
    wrapper: { ...wrapper, instance: () => instance }
  }
}
