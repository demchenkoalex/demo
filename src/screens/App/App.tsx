import React, { PureComponent } from 'react'
import { SafeAreaView, Text, TextStyle } from 'react-native'
import { UsersList } from 'src/components'
import styles from './styles'

export interface State {
  page?: number
}

export class App extends PureComponent<{}, State> {
  state = {
    page: undefined
  }

  handlePageChange = (page: number) => this.setState({ page })

  render () {
    const { page } = this.state

    return (
      <SafeAreaView style={styles.container}>
        <UsersList
          itemsPerPage={5}
          onPageChange={this.handlePageChange}
        />
        {page && <Text style={styles.text as TextStyle}>{`You are on a page ${page}`}</Text>}
      </SafeAreaView>
    )
  }
}
