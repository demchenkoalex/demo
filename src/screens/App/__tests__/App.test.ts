import { setupJestShallow } from 'src/utils/testConfig'
import { App } from '../App'

describe('App', () => {
  const props = {}

  describe('Shapshots', () => {
    it('with default props', () => {
      const { wrapper } = setupJestShallow({
        Component: App,
        props
      })

      expect(wrapper).toMatchSnapshot()
    })

    it('with default props and page in state', () => {
      const { wrapper } = setupJestShallow({
        Component: App,
        props
      })

      wrapper.instance().setState({ page: 1 })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    it('has handlePageChange defined', () => {
      const { wrapper } = setupJestShallow({
        Component: App,
        props
      })

      expect(typeof wrapper.instance().handlePageChange).toBe('function')
      const page = 1
      expect(wrapper.instance().state.page).toBe(undefined)
      expect(wrapper.instance().handlePageChange(1)).toBe(undefined)
      expect(wrapper.instance().state.page).toBe(1)
    })
  })
})
