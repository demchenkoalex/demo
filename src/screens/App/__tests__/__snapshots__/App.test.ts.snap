// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Shapshots with default props 1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <UsersList
    itemsPerPage={5}
    onPageChange={[Function]}
  />
</SafeAreaView>
`;

exports[`App Shapshots with default props and page in state 1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <UsersList
    itemsPerPage={5}
    onPageChange={[Function]}
  />
</SafeAreaView>
`;
